src/Umbraco.Infrastructure/HostedServices/ServerRegistration/InstructionProcessTask.cs

```diff
@@ -54,5 +54,15 @@ public override Task PerformExecuteAsync(object state)
 
             return Task.CompletedTask;
         }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (_messenger is IDisposable disposable)
+            {
+                disposable?.Dispose();
+            }
```

Comment:
Shouldn't this also use the same `_disposedValue` and `disposing` check?

```suggestion
            if (!_disposedValue)
            {
                if (disposing && _messenger is IDisposable disposable)
                {
                    disposable.Dispose();
                }
                
                _disposedValue = true;
            }
```

And the type pattern expression will ensure `disposable` can't be NULL, so the null-conditional operator can be safely removed here.